### PR TITLE
Les indemnités journalière ne sont un revenu d'activité que les 3 premiers mois.

### PR DESCRIPTION
**Pull request by fpagnoux**
* Consider some indemnités journalieres as revenus de remplacement in PPA (and RSA), according to the date of the arret de travail.
* Introduce
	* `rsa_indemnites_journalieres_hors_activite`
	* `rsa_indemnites_journalieres_activite`
	* `date_arret_de_travail`

### CODE REVIEW

File: openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py [hunk 1]
```diff
@@ -74,3 +73,9 @@ def function(self, simulation, period):
         )
 
         return period, result
+
+class date_arret_de_travail(Variable):
+    column = DateCol(default = nan)
```
Comment (benjello) [inline openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py]: You should put a default date that do not modify the results previous the introduction of the feature. Here I would use the equivalent of + infinity for dates.
@fpagnoux do you agree ?

**Comment by cbenz** [comment]
@fpagnoux @benjello I'm not sure if you received the previous message on this PR because I used the new github-reviews system! Sorry for the noise otherwise

File: openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py (hunk 1 above)
Comment (fpagnoux) [inline openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py]: Oh, sorry, I didn't notice this. Which api calls crash ? In mes-aides, it seems fine.
The usual default date in openfisca is `1970-01-01`, which is not not always satisfying.

In that case I needed to be able to know if the date was defined, so a default date is not really good enough (unless we put `3000-01-01`, but that's dirty').

Maybe we can try to replace `nan` by `datetime64('NaT'))` ? @benjello do you know if there is an equivalent of + infinity for dates in numpy ?

Comment (cbenz) [inline openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py]: In `rsa_indemnites_journalieres_activite` I see:

```
is_date_arret_de_travail_undefined = (date_arret_de_travail == datetime64('NaT'))
```

so perhaps the JSON serialization and parsing has to be updated.

Or use a "+infinity" date (ex: 2999-12-31) to be sure that by default the person is never in "arrêt de travail" situation.

Comment (cbenz) [inline openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py]: @fpagnoux as you can see in the traceback, the crash is in `/variables` endpoint.

Comment (cbenz) [inline openfisca_france/model/revenus/remplacement/indemnites_journalieres_securite_sociale.py]: In short we have to choose between 2 ways of representing "undefined" dates.

## use not-a-time values

Numpy provides `numpy.datetime64('NaT')` which means not-a-time.

- pros: it's clean
- cons:
  - each formula has the responsibility to check for `NaT` values, which can easily be forgotten
  - `NaT` is not JSON-serializable so we have to change JSON serializer to handle this value by (simply) removing the key from the JSON dict

## use -infinite or +infinite dates

- pros:
  - no need to change JSON serialization
  - no need to change formulas
  - input variables have to choose if their default value is `MIN_DATE` or `MAX_DATE`, and the `DateCol(default = XXX)` is mandatory and constants are exported from `openfisca_core`
- cons:
  - not clean

--

My choice is -infinite and +infinite dates.

**Pull request merged**